front/lib/api/csv.ts

```diff
@@ -1,5 +1,65 @@
 import { parse } from "csv-parse";
 
+const POSSIBLE_VALUES_MAX_LEN = 32;
+const POSSIBLE_VALUES_MAX_COUNT = 16;
+
+interface ColumnTypeInfo {
+  type: "string" | "number" | "boolean" | "enum";
+  possibleValues: string[] | null;
+}
+
+export interface CSVRow {
+  [key: string]: string;
+}
+
+/**
+ * Detect the type of a column based on its values.
```

Comment:
So we are kind of duplicating work that is being done in `core` for tables query right ?

Reply:
Yes indeed :/

Reply:
@fontanierh I don't think I should have tried to call Core for that and I think it's okay to have it duplicated here even tho not ideal to have some duplicated logic. Do you think it's a blocker?

Reply:
Def not a blocker, just not super satisfying. I'm trying to think of ways to dry this in the future but nothing obvious coming to mind

Reply:
Yes, it's not entirely satisfying, but this code path will become the primary route for all file uploads within the app. This includes both the conversation view and files uploaded to a data source from within the app.